crates/data-generator/src/blocks/output.rs

```diff
@@ -0,0 +1,628 @@
+            self.var_ordinal.into_iter().for_each(|ord| {
+                tokens.append(Punct::new('=', Spacing::Alone));
+                tokens.append(Literal::usize_unsuffixed(ord));
+            });
+
+            // ","
+            tokens.append(Punct::new(',', Spacing::Alone));
+        }
+    }
+}
+
+mod prop_default {
+    use proc_macro2::{Ident, TokenStream};
+    use quote::{quote, ToTokens};
+
+    use super::StringExt as _;
+    use crate::blocks::model::StateProperty;
+
+    #[derive(Debug)]
+    pub struct PropertyDefault {
+        pub enum_name: Ident, // WeepingVines
+        pub def_var_name: Ident,
+    }
+
+    impl From<&StateProperty> for PropertyDefault {
+        fn from(def_prop: &StateProperty) -> Self {
+            PropertyDefault {
+                enum_name: def_prop.prop_enum.to_ident(),
+                def_var_name: def_prop.variant.to_ident(),
+            }
+        }
+    }
+
+    impl ToTokens for PropertyDefault {
+        fn to_tokens(&self, tokens: &mut TokenStream) {
+            let enum_name = &self.enum_name;
+            let def_var_name = &self.def_var_name;
+
+            tokens.extend(quote! {
+                impl Default for #enum_name {
+                    fn default() -> Self {
+                        Self::#def_var_name
+                    }
+                }
+            });
+        }
+    }
+}
+
+mod prop_convert {
+    use proc_macro2::{Ident, Literal, TokenStream};
+    use quote::{quote, ToTokens};
+
+    use super::StringExt as _;
+    use crate::blocks::model::{BlockProperty, BlockPropertyVariant};
+
+    pub struct PropertyConvert {
+        pub prop_name: String, // HoneyLevel
+        pub prop_vars: Vec<BlockPropertyVariant>,
+    }
+
+    impl From<BlockProperty> for PropertyConvert {
+        fn from(prop: BlockProperty) -> Self {
+            Self {
+                prop_name: prop.name_pc,
+                prop_vars: prop.variants,
+            }
+        }
+    }
+
+    impl ToTokens for PropertyConvert {
+        fn to_tokens(&self, tokens: &mut TokenStream) {
+            let prop_enum_name = self.prop_name.to_ident();
+            match self.prop_vars.iter().next() {
+                Some(BlockPropertyVariant::Bool(_, _)) => {
+                    PropertyNotBool(&prop_enum_name).to_tokens(tokens);
+                    PropertyFromBool(&prop_enum_name).to_tokens(tokens);
+                    PropertyIntoBool(&prop_enum_name).to_tokens(tokens);
+                }
+                Some(BlockPropertyVariant::Numeric(_, _)) => {
+                    let impl_from_u8 = PropertyFromU8 {
+                        enum_name: &prop_enum_name,
+                        vars: &self.prop_vars,
+                    };
+                    impl_from_u8.to_tokens(tokens);
+                }
+                _ => {}
+            }
+        }
+    }
+
+    pub struct PropertyFromU8<'a> {
+        pub enum_name: &'a Ident,
+        pub vars: &'a Vec<BlockPropertyVariant>,
```

Comment:
```suggestion
        pub vars: &'a [BlockPropertyVariant],
```

I don't think that you gain anything helpful from non-`mut` ref to `Vec` instead of slice.